### PR TITLE
Search: pass the 'explain' flag from the UI and debug paging issues

### PR DESCRIPTION
**Pull request by ryantxu**
This PR is to help debug potential paging issues with the new search engine and infinite scroll.  However, I am unable to replicate a real issue.  To test add "&explain=true" to the URL.  IE:
http://localhost:3000/?orgId=1&search=open&query=&explain=true

### CODE REVIEW

File: pkg/services/searchV2/bluge.go [hunk 1]
```diff
@@ -600,7 +601,15 @@ func doSearchQuery(
 		}
 
 		if q.Explain {
-			fScore.Append(match.Score)
+			if isMatchAllQuery || true {
+				fScore.Append(float64(fieldLen + q.From))
+
+				// HACK, just for helping test paging issue
+				current := fName.At(fieldLen)
+				fName.Set(fieldLen, fmt.Sprintf("[%d/%d] %s", q.From, fieldLen, current))
```
Comment (ryantxu) [inline pkg/services/searchV2/bluge.go:609]: DO NOT COMMIT!  (but easier to debug then looking at the request!)

File: public/app/features/search/service/bluge.ts [hunk 2]
```diff
@@ -141,7 +141,7 @@ async function doSearchQuery(query: SearchQuery): Promise<QueryResponse> {
                 search: {
                   ...(target?.search ?? {}),
                   from,
-                  limit: Math.max(limit, nextPageSizes),
+                  limit: Math.min(limit, nextPageSizes),
```
Comment (ryantxu) [inline public/app/features/search/service/bluge.ts:144]: this *may* be the issue you were seeing @ArturWierzbicki, with bigger blocks it should not really matter though

File: public/app/features/search/service/bluge.ts [hunk 3]
```diff
@@ -69,8 +69,8 @@ export class BlugeSearcher implements GrafanaSearcher {
   }
 }
 
-const firstPageSize = 50;
-const nextPageSizes = 100;
+const firstPageSize = 5; //50;
```
Comment (ryantxu) [inline public/app/features/search/service/bluge.ts:72]: The UI expects the first query to return enough results to fill the screen, so this is weird -- but does make it easier to debug :)

File: public/app/features/search/service/bluge.ts [hunk 4]
```diff
@@ -121,15 +121,13 @@ async function doSearchQuery(query: SearchQuery): Promise<QueryResponse> {
...
-    loadMoreItems: async (startIndex: number, stopIndex: number): Promise<void> => {
-      console.log('LOAD NEXT PAGE', { startIndex, stopIndex, length: view.dataFrame.length });
+  let loadMax = 0;
+  let pending: Promise<void> | undefined = undefined;
+  const getNextPage = async () => {
+    while (loadMax > view.dataFrame.length) {
       const from = view.dataFrame.length;
-      const limit = stopIndex - from;
-      if (limit < 0) {
+      console.log('Load NEXT PAGE', { from, loadMax });
```
Comment (ArturWierzbicki) [inline public/app/features/search/service/bluge.ts:129]: should we either remove or maybe hide both these `console.log`s behind the `explain` query param?

**Review by ArturWierzbicki — APPROVED** [review]
I can't reproduce the original issue - niceee 🚀🚀

**Pull request merged**